internal/services/dispatcher/server_v1.go

```diff
@@ -27,6 +27,194 @@ import (
 	tasktypes "github.com/hatchet-dev/hatchet/internal/services/shared/tasktypes/v1"
 )
 
+type StreamEventBuffer struct {
+	stepRunIdToWorkflowEvents map[string][]*contracts.WorkflowEvent
+	stepRunIdToExpectedIndex  map[string]int64
+	stepRunIdToLastSeenTime   map[string]time.Time
+	stepRunIdToCompletionTime map[string]time.Time
+	mu                        sync.RWMutex
```

Comment:
Looks like read locking in `RWMutex` isn't used here so more straightforward to just use `Mutex`